### DISCUSSION TITLE
Help with creating custom pages plugin

### DISCUSSION THREAD

**Discussion by dmitrijsmihailovs** [filamentphp/filament#1632] [Help]
Hi, I have started to work on creating Pages plugin.
Basic idea is that there is Page model with name, slug, nested set, morph relation fields etc.
When you want to create a page, you select template, which is a model, for example, BlogTemplate and for each template there could be different fields - text, image, rich editor etc
At the moment I'm stuck at trying to resolve these fields from templates in PageResource form, because I can't get the query string value there, but as far as I understand I can only get it in CreateRecord/EditRecord pages
And I'm not sure how I could get model in edit so I can get template from it

**Comment by danharrin** [comment]
Hmm, could you resolve query string values from `request()->query()`? If not, you can access the Livewire component (edit / create page) from any form component by defining the `$livewire` property of a closure.

https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

**Comment by dmitrijsmihailovs** [comment]
Had some time to continue working on this. Saw the new Fieldset field with relationship and tried it to use it.
I have a Node model with relationship

`public function content(): MorphTo
    {
        return $this->morphTo();
    }`

and this form in the NodeResource

`
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                    ->required()
                    ->reactive()
                    ->afterStateUpdated(function (Closure $set, $state) {
                        $set('slug', Str::slug($state));
                    }),
                TextInput::make('slug')
                    ->required()
                    ->unique(Node::class, 'slug', fn($record) => $record),
                Fieldset::make('Content')
                    ->relationship('content')
                    ->schema([
                        TextInput::make('text')
                    ])
            ]);
    }
`
    
And I get Call to a member function content() on null
From what I can see, model is null for some reason
